Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -114,6 +114,21 @@ public void SerializesAndDeserializesSubtypesBasedOnTypeAnnotations()
             );
         }
 
+        [TestMethod]
+        public void DeserializesObjectsThatExceedMaxDepth()
+        {
+            var stream = new IonSerializer().Serialize(TestObjects.UnitedStates);
+            
+            var serializer = new IonSerializer(new IonSerializationOptions {MaxDepth = 5});
+            var deserialized = serializer.Deserialize<Country>(stream);
+            
+            Assert.IsNotNull(deserialized.States);
+            Assert.IsNotNull(deserialized.States[0].Capital);
+            Assert.IsNotNull(deserialized.States[0].Capital.Mayor);
+            Assert.IsNotNull(deserialized.States[0].Capital.Mayor.Party);
+            Assert.IsNull(deserialized.States[0].Capital.Mayor.Party.Name);
```

Comment:
Lines 125-127 are redundant. They cannot be null if 128 is not null.

Reply:
Removed redundant asserts.